### PR TITLE
#162412611 user can signin endpoint

### PR DESCRIPTION
**Pull request by bl4ck4ndbr0wn**
#### What does this PR do?
Have Post endpoint to allow users to login or signin
#### Description of Task to be completed?
- Test POST /api/v2/auth/login and make them fail
- Create login endpoints 
- Test API endpoint to confirm that the register is working.
#### How should this be manually tested?
Clone the repo as bellow:
 ```
 git clone - b ft-user-signin-162412611 git@github.com: bl4ck4ndbr0wn/iReporter.git
```
Create a virtual environment and activate it
 ```
 python3 - m venv env
 source .env
 ```
 Install all the dependencies needed by
 ```
 pip install - r requirements.txt
 ```
 Run the flask application
 ```
  flask run
 ```
On postname try sending a request
```
* POST  /api/v2/auth/login a user
* :params
      {  
         “username” : String,
         “password” : String,
       }
```
#### Any background context you want to provide?
Create a postgresql database and name it ireporter.
#### What are the relevant pivotal tracker stories?
 [  # 162412611](https://www.pivotaltracker.com/n/projects/2226962)

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -3,12 +3,6 @@
 from flask_jwt_extended import JWTManager
```
Comment (thoth-ky) [inline app/__init__.py]: Do away with all jwt-extended stuff

File: app/api/v2/models/user.py [hunk 2]
```diff
@@ -37,7 +39,8 @@ def find_by_name(self, username):
         :type username: str
         :return: User instance
         """
-        self.cursor.execute("SELECT * FROM users WHERE username=%s", (username,))
+        self.cursor.execute("SELECT * FROM users "
+                            "WHERE username=%s", (username,))
```
Comment (thoth-ky) [inline app/api/v2/models/user.py]: username is an undefined variable here

Comment (thoth-ky) [reply app/api/v2/models/user.py]: you can use `self.username` instead

File: app/tests/v2/system/test_user.py [hunk 3]
```diff
@@ -1,6 +1,102 @@
...
+        """
+        response = self.signup()
+        self.assertEqual(response.status_code, 201)
+        self.assertDictEqual({"status": 201,
+                              "data": [{
+                                    "message": "User created Successfully."
+                              }]},
+                             json.loads(response.data))
+
+    def test_user_register_User_exist(self):
```
Comment (thoth-ky) [inline app/tests/v2/system/test_user.py]:
```suggestion
    def test_user_registration_with_duplicate_usernames(self):
```

File: app/tests/v2/system/test_user.py [hunk 4]
```diff
@@ -1,6 +1,102 @@
...
+        """
+        Test login.
+        :return Success status
+        """
+        response = self.login()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertIsNotNone(User.find_by_name("alpha"))
+
+    def test_user_login_user_not_exist(self):
```
Comment (thoth-ky) [inline app/tests/v2/system/test_user.py]:
```suggestion
    def test_unregistered_user_can_not_login(self):
```

File: app/tests/v2/system/test_user.py [hunk 5]
```diff
@@ -1,6 +1,102 @@
...
+                                  {
+                                      "message": "A user with that"
+                                                 " username doesn't exists"
+                                  }
+                              ]},
+                             json.loads(response.data),
+                             "The response data is not the same."
+                             )
+
+    def test_get_user_token(self):
```
Comment (thoth-ky) [inline app/tests/v2/system/test_user.py]:
```suggestion
    def test_user_gets_token_on_login(self):
```

**Review by thoth-ky — COMMENTED** [review]
I suggest you try naming your tests with names that give them more context. This makes documentation easier

File: app/tests/v2/system/test_user.py (hunk 4 above)
Comment (bl4ck4ndbr0wn) [reply app/tests/v2/system/test_user.py]: done

File: app/tests/v2/system/test_user.py (hunk 5 above)
Comment (bl4ck4ndbr0wn) [reply app/tests/v2/system/test_user.py]: done

File: app/api/v2/models/user.py (hunk 2 above)
Comment (bl4ck4ndbr0wn) [reply app/api/v2/models/user.py]: done

File: app/__init__.py (hunk 1 above)
Comment (bl4ck4ndbr0wn) [reply app/__init__.py]: done

**Review by thoth-ky — APPROVED** [review]
looks good you can merge

**Pull request merged**